lib/topics.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+const assert = require('assert');
+const SNS    = require('./sns');
+
+
+// Abstraction for a topic.
+class Topic {
+
+  constructor(server, topicName) {
+    this._server = server;
+    this.name    = topicName;
+  }
+
+
+  // Publish to a topic.
+  publish(message) {
+    return this.clientPromise
+      .then(client => {
+        const stringMessage = JSON.stringify(message);
+        return client.publish(this.name, stringMessage);
+      });
+  }
+
+
+  // -- Clients --
+
+  get clientPromise() {
+    if (!this._clientPromise)
+      this._clientPromise = this._server.configPromise.then(client => this._clientFromConfig(client));
+    return this._clientPromise;
+  }
+
+  _clientFromConfig(config) {
+    switch (config.service) {
+      case 'aws': {
+        const awsConfig = {
+          accessKeyId:     config.accessKeyId,
+          secretAccessKey: config.secretAccessKey,
+          region:          config.region,
+          topicName:       config.prefixedName(this.name)
+        };
+        return new SNS(awsConfig);
+      }
+
+      default:
+        throw new Error('Pub/sub only supported in AWS.');
+    }
+  }
+
+}
+
+
+// Abstracts the topic server.
+module.exports = class Topics {
+
+  constructor(ironium) {
+    this._ironium = ironium;
+    this._topics  = new Map();
+  }
+
+  // Returns the named queue, queue created on demand.
+  getTopic(topicName) {
+    assert(topicName, 'Missing topic name');
+
+    if (!this._topics.has(topicName)) {
+      const topic = new Topic(this, topicName);
+
+      if (this.started)
+        topic.start();
```

Comment:
We should be doing subscribing. That would  answer "what testing facility we're going to provide" — you subscribe and check that you received the notification. You receive a notification, not poll for notification, so there's no mechanism to start. It would be better to do that at some other time, i.e first figure out SQS which also involves some changes to scheduling. One problem at a time.